### PR TITLE
feat: tighten check_agent_safe_scope.py edge cases (closes #90)

### LINKED ISSUE #90: Tighten check_agent_safe_scope.py edge cases (bare top-level paths + taxonomy ground-truthing)
## Context

PR #83 (closes #77) introduced `scripts/check_agent_safe_scope.py` as the mechanical scope-boundary check for agent-safe PRs. Two edge cases were surfaced during PR #83's review cycle but not addressed in that PR (deferred to keep PR #83's scope clean).

### Finding 1: bare top-level paths drop silently

`parse_files_to_touch()`'s bare-path tokenization filters tokens through:

```python
looks_like_path = "/" in token or token in {
    "CLAUDE.md", "README.md", "CHANGELOG.md",
}
```

Any top-level file not in the hardcoded allowed set is silently dropped. If an issue body's "Files to touch" section lists `pyproject.toml` or `tasks.py` as a bare path (no backticks), it's filtered out, so a PR touching that file would FAIL even though the issue body explicitly authorised it.

Backticked paths (the canonical form) are unaffected — they're captured by the earlier `_BACKTICK_TOKEN_RE` branch.

### Finding 2: taxonomy maps are forward-looking, not ground-truthed

`BOUNDED_AREA_GLOBS` and `META_AREAS` include entries that are not yet live labels (`mcp`, `sdk` in BOUNDED_AREA_GLOBS; most of META_AREAS). A forward-looking explanatory comment was added in commit b51f28b on PR #83, but the maps haven't been ground-truthed against the live label set.

## Proposal

Two fixes, both bounded:

### Fix 1: Extend bare-path acceptance

Either:
- (a) Extend the allowed top-level set to include `pyproject.toml`, `tasks.py`, `Makefile`, and any other top-level file that legitimately appears in `Files to touch` sections
- (b) Relax the `looks_like_path` heuristic to accept any token matching `\.\w+$` (has a file extension)
- (c) Document explicitly that bare top-level paths must be backticked, and add a parser warning when an unbacticked token in a bullet looks like it could be a top-level file

Design-review picks. Lean (b) for permissiveness or (c) for strictness.

### Fix 2: Ground-truth taxonomy maps

Compare BOUNDED_AREA_GLOBS keys + META_AREAS members against the live label taxonomy via:
```bash
gh api /repos/warlordofmars/agentcore-starter/labels --paginate \
  --jq '.[].name' | sort
```

Resolve discrepancies:
- Remove map entries that don't correspond to live labels (or keep them with explicit "forward-looking, not yet live" documentation)
- Add live labels not currently mapped

## Acceptance criteria

- [ ] Bare-path tokenization handles top-level files appearing in Files-to-touch (decision documented for which approach above)
- [ ] BOUNDED_AREA_GLOBS keys exactly match the bounded subset of live area labels OR have explicit per-entry docstrings explaining why an entry is forward-looking
- [ ] META_AREAS exactly matches the meta subset of live area labels OR same documentation requirement
- [ ] Unit tests cover both fixes
- [ ] If approach (c) is taken, the parser warning is exercised by a test

## Notes

- agent-safe (mechanical scope, scoped to two files: `scripts/check_agent_safe_scope.py` and `tests/unit/test_check_agent_safe_scope.py`)
- Source: PR #83 review-cycle findings 2 and 3
- Lower priority than #50 / Issue 2 of this batch — these are edge cases the gate handles imperfectly, not load-bearing failures

### PR DESCRIPTION
**Pull request by warlordofmars**
Closes #90

## Summary

Two mechanical fixes to `scripts/check_agent_safe_scope.py`, per the design locked on the issue 2026-04-26:

- **Fix 1 — bare top-level path acceptance**: relax the no-backtick path heuristic from a hard-coded allowlist (`CLAUDE.md`, `README.md`, `CHANGELOG.md`) to "contains `/` OR ends in a file extension". Bare tokens like `tasks.py`, `pyproject.toml`, `uv.lock` now parse as paths without needing backticks. Edge case: tokens with no `/` and no extension (e.g. `Makefile`) still drop — the canonical workaround (backticks) is now documented in the parser docstring.
- **Fix 2 — ground-truth taxonomy maps**: prune `BOUNDED_AREA_GLOBS` (drop `mcp`, `sdk`) and `META_AREAS` (drop `compliance`, `marketing`, `growth`, `seo`, `design`, `ops`, `performance`, `ux`, `a11y`) so both maps reflect the live label set. Rewrite the top-comment to drop the "forward-looking" note and document the canonical refresh procedure (`gh api /repos/.../labels --paginate --jq '.[].name'`).

Two new test cases per fix exercise both branches of the new disjunction (Fix 1) and confirm pruned keys no longer map while kept keys still resolve (Fix 2). All 41 tests pass; pre-push gate green.

## Approach

- Followed the design comment verbatim — option (b) for Fix 1, full prune for Fix 2.
- Acceptance criterion 5 from the issue body (parser-warning test for ambiguous tokens) is moot under option (b) and was deliberately not implemented; the design comment overrides the AC.
- Local Copilot review surfaced one finding: `CLAUDE.md` §Area lines 558-560 still lists the pruned labels (`mcp`, `sdk`, `ux`, `a11y`, `compliance`, `design`, `performance`, `marketing`, `seo`, `growth`, `ops`). This is a **deliberate, documented divergence** — the existing test `test_area_label_paths_bounded_area_documentation` already notes "CLAUDE.md taxonomy is forward-looking; the live label set is the ground truth", and the design comment scopes this PR to the script + its tests only. CLAUDE.md taxonomy reconciliation is a separate concern (file a follow-up if desired).

### CODE REVIEW

File: scripts/check_agent_safe_scope.py [hunk 1]
```diff
@@ -178,16 +181,15 @@ def parse_files_to_touch(body: str) -> list[str] | None:
...
                 continue
-            looks_like_path = "/" in token or token in {
-                "CLAUDE.md",
-                "README.md",
-                "CHANGELOG.md",
-            }
+            # A token looks like a path if it contains `/` OR ends in a file
+            # extension. Tokens with no extension and no `/` (e.g. `Makefile`)
+            # are dropped — wrap them in backticks for explicit acceptance.
+            looks_like_path = "/" in token or bool(re.search(r"\.\w+$", token))
```
[the remark this answers is not in the document]
Comment (warlordofmars) [reply scripts/check_agent_safe_scope.py]: Fixed in f898eb3224c61aa621c9191dd2264f4be3388902 — added a path-safe character filter (`re.fullmatch(r"[\w./@\-+]+", token)`) before the extension check. Prose fragments like `(e.g.` or `v1.0)` are now rejected because they contain parens; real paths still pass. New regression test `test_fix1_prose_fragments_with_dots_are_not_treated_as_paths` covers the realistic `- Edit: src/foo.py (e.g. for handling auth)` case.

File: scripts/check_agent_safe_scope.py [hunk 2]
```diff
@@ -178,16 +184,23 @@ def parse_files_to_touch(body: str) -> list[str] | None:
...
-            }
+            # Path-safe character filter: only word chars, dot, slash, hyphen,
+            # plus, at-sign, underscore (already in \w). Anything else (parens,
+            # quotes, colons, brackets) marks the token as prose, not a path.
+            if not re.fullmatch(r"[\w./@\-+]+", token):
+                continue
+            # A token looks like a path if it contains `/` OR ends in a file
+            # extension. Tokens with no extension and no `/` (e.g. `Makefile`)
+            # are dropped — wrap them in backticks for explicit acceptance.
+            looks_like_path = "/" in token or bool(re.search(r"\.\w+$", token))
```
[the remark this answers is not in the document]
Comment (warlordofmars) [reply scripts/check_agent_safe_scope.py]: Fixed in 3df640c2e159e570d10eb68c7f2f4ce379afdc4d — tightened the extension regex to `\.[A-Za-z]\w*$` (first char of extension must be a letter). Version-like tokens `v1.0`, `2.10` no longer match because `.0` and `.10` start with digits. Real paths (`tasks.py`, `pyproject.toml`, `package-lock.json`) all start with a letter so still pass. New regression test `test_fix1_version_like_tokens_are_not_treated_as_paths` covers the case.

File: scripts/check_agent_safe_scope.py [hunk 3]
```diff
@@ -178,16 +186,26 @@ def parse_files_to_touch(body: str) -> list[str] | None:
...
+            # quotes, colons, brackets) marks the token as prose, not a path.
+            if not re.fullmatch(r"[\w./@\-+]+", token):
+                continue
+            # A token looks like a path if it contains `/` OR ends in a file
+            # extension whose first character is a letter (so version-like
+            # tokens such as `v1.0` are NOT treated as paths — those would
+            # match `\.\w+$` because `\w` includes digits). Tokens with no
+            # extension and no `/` (e.g. `Makefile`) are dropped — wrap them
+            # in backticks for explicit acceptance.
+            looks_like_path = "/" in token or bool(re.search(r"\.[A-Za-z]\w*$", token))
```
[the remark this answers is not in the document]
Comment (warlordofmars) [reply scripts/check_agent_safe_scope.py]: Fixed in 51b27682e864178b3ff3a9284b9e9a3ff1975aba — applied your suggested tightening: regex is now `\.[A-Za-z]\w+$` (>=2 chars after the dot). `e.g` and `i.e` no longer match. Trade-off: real single-letter extensions `.c`, `.r` also drop from the no-backtick fallback; backticks remain the workaround, and none exist in this repo. New regression test `test_fix1_unparenthesised_abbreviations_are_not_treated_as_paths` covers the case.

File: tests/unit/test_check_agent_safe_scope.py [hunk 4]
```diff
@@ -424,3 +424,171 @@ def test_universal_allowed_changelog_passes_in_files_to_touch_mode():
...
+    assert not any("e.g" in t for t in parsed)
+    assert not any(t.startswith("v1.") for t in parsed)
+
+
+def test_fix1_version_like_tokens_are_not_treated_as_paths():
+    """Issue #90 Fix 1 hardening (Copilot iter 2): tokens like `v1.0`,
+    `v1.2`, or `2.10` would pass the path-safe character filter (only word
+    chars + dot) AND match `\\.\\w+$` because `\\w` includes digits. The
+    extension regex is tightened to require the first extension character
+    to be a letter (`\\.[A-Za-z]\\w*$`), rejecting version-like tokens.
```
[the remark this answers is not in the document]
Comment (warlordofmars) [reply tests/unit/test_check_agent_safe_scope.py]: Fixed in 9d709db782ddd784f7848f251b2467be4c68d712 — docstring updated to reference `\.[A-Za-z]\w+$` (matching the implementation post-iter-3 fix).
